Remove unused code

### Summary 
There are functions in the code base, which are not used anywhere, and can be removed. 
Sample when running
`GO111MODULE=on golangci-lint run --no-config --disable-all -E unused ./...`
```
...
pkg/github/github_app.go:28:22: func `(*GithubApp).isGithubAppEnabled` is unused (unused)
pkg/cmd/step/git/step_git.go:51:26: func `(*StepGitOptions).dropRepositories` is unused (unused)
pkg/cmd/get/get_build_logs.go:469:31: func `(*GetBuildLogsOptions).loadPipelineActivities` is unused (unused)
pkg/cmd/step/step_next_version.go:427:34: func `(*StepNextVersionOptions).setChartVersion` is unused (unused)
pkg/helm/helm_template.go:1057:24: func `(*HelmTemplate).getChartNameAndVersion` is unused (unused)
...
```

--- a/pkg/builds/jenkinsfiles.go
+++ b/pkg/builds/jenkinsfiles.go
@@ -120,11 +120,6 @@ func (j *JenkinsConverter) println(text string) {
 	j.writeString("\n")
 }
 
-func (j *JenkinsConverter) printf(format string, args ...interface{}) {
-	j.writeIndent()
-	j.writeString(fmt.Sprintf(format, args...))
-}
-
 func (j *JenkinsConverter) String() string {
 	j.writer.Flush()
 	return j.buffer.String()

--- a/pkg/cmd/controller/controller_enviornmentcontroller.go
+++ b/pkg/cmd/controller/controller_enviornmentcontroller.go
@@ -429,19 +429,6 @@ func (o *ControllerEnvironmentOptions) isReady() bool {
 	return true
 }
 
-func (o *ControllerEnvironmentOptions) unmarshalBody(w http.ResponseWriter, r *http.Request, result interface{}) error {
-	// TODO assume JSON for now
-	data, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		return errors.Wrap(err, "reading the JSON request body")
-	}
-	err = json.Unmarshal(data, result)
-	if err != nil {
-		return errors.Wrap(err, "unmarshalling the JSON request body")
-	}
-	return nil
-}
-
 func (o *ControllerEnvironmentOptions) marshalPayload(w http.ResponseWriter, r *http.Request, payload interface{}) error {
 	data, err := json.Marshal(payload)
 	if err != nil {

--- a/pkg/cmd/create/create_cluster_iks.go
+++ b/pkg/cmd/create/create_cluster_iks.go
@@ -3,7 +3,6 @@ package create
 import (
 	"math"
 	"os"
-	"regexp"
 	"strconv"
 	"strings"
 	"time"
@@ -84,23 +83,7 @@ var (
 		jx create cluster iks
 
 `)
-	re = regexp.MustCompile("^([0-9]+)")
 )
-
-type byNumberIndex []string
-
-func (s byNumberIndex) Len() int {
-	return len(s)
-}
-func (s byNumberIndex) Swap(i, j int) {
-	s[i], s[j] = s[j], s[i]
-}
-func (s byNumberIndex) Less(i, j int) bool {
-	var left, right int = 0, 0
-	left, _ = strconv.Atoi(re.FindAllString(s[i], -1)[0])
-	right, _ = strconv.Atoi(re.FindAllString(s[j], -1)[0])
-	return left < right
-}
 
 // NewCmdGet creates a command object for the generic "init" action, which
 // installs the dependencies required to run the jenkins-x platform on a kubernetes cluster.

--- a/pkg/cmd/get/get_apps.go
+++ b/pkg/cmd/get/get_apps.go
@@ -1,7 +1,6 @@
 package get
 
 import (
-	"encoding/json"
 	"fmt"
 
 	v1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
@@ -177,15 +176,6 @@ func (o *GetAppsOptions) Run() error {
 	return nil
 }
 
-func (o *GetAppsOptions) generateAppStatusOutput(app *v1.App) error {
-	name := app.Labels[helm.LabelReleaseName]
-	output, err := o.Helm().StatusReleaseWithOutput(o.Namespace, name, "json")
-	if err != nil {
-		return err
-	}
-	return o.printHelmResourcesWithFormat(output)
-}
-
 func (o *GetAppsOptions) generateTableFormatted(apps *v1.AppList) appsResult {
 	releases, err := o.getAppsStatus(o.GitOps, o.Namespace, apps)
 	if err != nil {
@@ -247,20 +237,6 @@ func (o *GetAppsOptions) generateTableHeaders(apps *v1.AppList) table.Table {
 	titles := []string{"Name", "Version", "Chart Repository", "Namespace", "Status", "Description"}
 	t.AddRow(titles...)
 	return t
-}
-
-func (o *GetAppsOptions) printHelmResourcesWithFormat(helmOutputJSON string) error {
-	h := HelmOutput{}
-	err := json.Unmarshal([]byte(helmOutputJSON), &h)
-	if err != nil {
-		return err
-	}
-	if o.Output == "" {
-		fmt.Fprintln(o.Out, h.helmInfoStatus.Resources)
-		return nil
-	}
-	return o.renderResult(h, o.Output)
-
 }
 
 func (o *GetAppsOptions) getAppsStatus(gitOps bool, namespace string, apps *v1.AppList) (map[string]string, error) {

--- a/pkg/cmd/get/get_build_logs.go
+++ b/pkg/cmd/get/get_build_logs.go
@@ -6,17 +6,14 @@ import (
 	"strings"
 	"time"
 
-	v1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
 	"github.com/jenkins-x/jx/pkg/builds"
 	"github.com/jenkins-x/jx/pkg/client/clientset/versioned"
 	"github.com/jenkins-x/jx/pkg/cmd/helper"
 	"github.com/jenkins-x/jx/pkg/gits"
 	"github.com/jenkins-x/jx/pkg/kube"
 	"github.com/jenkins-x/jx/pkg/logs"
-	"github.com/jenkins-x/jx/pkg/tekton"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/jenkins-x/jx/pkg/cmd/opts"
@@ -24,8 +21,6 @@ import (
 	"github.com/jenkins-x/jx/pkg/log"
 	"github.com/jenkins-x/jx/pkg/util"
 	tektonclient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // GetBuildLogsOptions the command line options
@@ -271,99 +266,4 @@ func (o *CLILogWriter) WriteLog(logLine logs.LogLine, lch chan<- logs.LogLine) e
 func (o *CLILogWriter) BytesLimit() int {
 	//We are not limiting bytes with this writer
 	return 0
-}
-
-// loadPipelines loads all available pipelines as PipelineRunInfos.
-func (o *GetBuildLogsOptions) loadPipelines(kubeClient kubernetes.Interface, tektonClient tektonclient.Interface, jxClient versioned.Interface, ns string) ([]string, string, map[string][]builds.BaseBuildInfo, error) {
-	defaultName := ""
-	names := []string{}
-	pipelineMap := map[string][]builds.BaseBuildInfo{}
-
-	labelSelectors := o.BuildFilter.LabelSelectorsForBuild()
-
-	listOptions := metav1.ListOptions{}
-	if len(labelSelectors) > 0 {
-		listOptions.LabelSelector = strings.Join(labelSelectors, ",")
-	}
-
-	prList, err := tektonClient.TektonV1alpha1().PipelineRuns(ns).List(listOptions)
-	if err != nil && !apierrors.IsNotFound(err) {
-		log.Logger().Warnf("Failed to query PipelineRuns %s", err)
-		return names, defaultName, pipelineMap, err
-	}
-
-	structures, err := jxClient.JenkinsV1().PipelineStructures(ns).List(listOptions)
-	if err != nil && !apierrors.IsNotFound(err) {
-		log.Logger().Warnf("Failed to query PipelineStructures %s", err)
-		return names, defaultName, pipelineMap, err
-	}
-	// TODO: Remove this eventually - it's only here for structures created before we started applying labels to them in v2.0.216.
-	if len(prList.Items) > len(structures.Items) && len(labelSelectors) != 0 {
-		structures, err = jxClient.JenkinsV1().PipelineStructures(ns).List(metav1.ListOptions{})
-		if err != nil && !apierrors.IsNotFound(err) {
-			log.Logger().Warnf("Failed to query PipelineStructures %s", err)
-			return names, defaultName, pipelineMap, err
-		}
-	}
-
-	buildInfos := []*tekton.PipelineRunInfo{}
-
-	podLabelSelector := pipeline.GroupName + pipeline.PipelineRunLabelKey
-	if len(labelSelectors) > 0 {
-		podLabelSelector += "," + strings.Join(labelSelectors, ",")
-	}
-	podList, err := kubeClient.CoreV1().Pods(ns).List(metav1.ListOptions{
-		LabelSelector: podLabelSelector,
-	})
-	if err != nil {
-		return names, defaultName, pipelineMap, err
-	}
-	for _, pr := range prList.Items {
-		var ps v1.PipelineStructure
-		for _, p := range structures.Items {
-			if p.Name == pr.Name {
-				ps = p
-			}
-		}
-		pri, err := tekton.CreatePipelineRunInfo(pr.Name, podList, &ps, &pr)
-		if err != nil {
-			log.Logger().Warnf("Error creating PipelineRunInfo for PipelineRun %s: %s", pr.Name, err)
-		}
-		if pri != nil && o.BuildFilter.BuildMatches(pri.ToBuildPodInfo()) {
-			buildInfos = append(buildInfos, pri)
-		}
-	}
-
-	tekton.SortPipelineRunInfos(buildInfos)
-	if len(buildInfos) == 0 {
-		return names, defaultName, pipelineMap, fmt.Errorf("no Tekton pipelines have been triggered which match the current filter")
-	}
-
-	namesMap := make(map[string]bool, 0)
-	for _, build := range buildInfos {
-		buildName := build.Pipeline + " #" + build.Build
-		if build.Context != "" {
-			buildName += " " + build.Context
-		}
-		namesMap[buildName] = true
-		pipelineMap[buildName] = append(pipelineMap[buildName], build)
-
-		if build.Branch == "master" {
-			defaultName = buildName
-		}
-	}
-	for k := range namesMap {
-		names = append(names, k)
-	}
-
-	return names, defaultName, pipelineMap, nil
-}
-
-func (o *GetBuildLogsOptions) loadPipelineActivities(jxClient versioned.Interface, ns string) (*v1.PipelineActivityList, error) {
-	paList, err := jxClient.JenkinsV1().PipelineActivities(ns).List(metav1.ListOptions{})
-	if err != nil {
-		return nil, errors.Wrap(err, "there was a problem getting the PipelineActivities")
-	}
-
-	return paList, nil
 }

--- a/pkg/cmd/get/get_issues.go
+++ b/pkg/cmd/get/get_issues.go
@@ -1,13 +1,10 @@
 package get
 
 import (
-	"strings"
-
 	"github.com/jenkins-x/jx/pkg/cmd/helper"
 
 	"github.com/spf13/cobra"
 
-	gojenkins "github.com/jenkins-x/golang-jenkins"
 	"github.com/jenkins-x/jx/pkg/cmd/opts"
 	"github.com/jenkins-x/jx/pkg/cmd/templates"
 )
@@ -77,18 +74,4 @@ func (o *GetIssuesOptions) Run() error {
 	}
 	table.Render()
 	return nil
-}
-
-func (o *GetIssuesOptions) matchesFilter(job *gojenkins.Job) bool {
-	args := o.Args
-	if len(args) == 0 {
-		return true
-	}
-	name := job.FullName
-	for _, arg := range args {
-		if strings.Contains(name, arg) {
-			return true
-		}
-	}
-	return false
 }

--- a/pkg/cmd/initcmd/init.go
+++ b/pkg/cmd/initcmd/init.go
@@ -609,14 +609,6 @@ controller:
 	return nil
 }
 
-func (o *InitOptions) ingressNamespace() string {
-	ingressNamespace := "kube-system"
-	if !o.Flags.GlobalTiller {
-		ingressNamespace = o.Flags.Namespace
-	}
-	return ingressNamespace
-}
-
 // ValidateGit validates that git is configured correctly
 func (o *InitOptions) ValidateGit() error {
 	// lets ignore errors which indicate no value set

--- a/pkg/cmd/step/boot/step_boot_vault.go
+++ b/pkg/cmd/step/boot/step_boot_vault.go
@@ -288,18 +288,6 @@ func (o *StepBootVaultOptions) installOperator(requirements *config.Requirements
 	return nil
 }
 
-func (o *StepBootVaultOptions) createIngressConfig(requirements *config.RequirementsConfig) (kube.IngressConfig, error) {
-	i := requirements.Ingress
-	tls := i.TLS
-	ic := kube.IngressConfig{
-		Domain:  i.Domain,
-		Exposer: "Ingress",
-		Email:   tls.Email,
-		TLS:     tls.Enabled,
-	}
-	return ic, nil
-}
-
 // verifyVaultIngress verifies there is a Vault ingress and if not create one if there is a file at
 func (o *StepBootVaultOptions) verifyVaultIngress(requirements *config.RequirementsConfig, kubeClient kubernetes.Interface, ns string, systemVaultName string) (bool, error) {
 	fileName := filepath.Join(o.Dir, "vault-ing.tmpl.yaml")

--- a/pkg/cmd/step/create/step_create_task.go
+++ b/pkg/cmd/step/create/step_create_task.go
@@ -1445,25 +1445,6 @@ func (o *StepCreateTaskOptions) invokeSteps(steps []*syntax.Step) error {
 	return nil
 }
 
-func (o *StepCreateTaskOptions) dockerImage(projectConfig *config.ProjectConfig, gitInfo *gits.GitRepository) string {
-	dockerRegistry := o.getDockerRegistry(projectConfig)
-
-	dockerRegistryOrg := o.DockerRegistryOrg
-	if dockerRegistryOrg == "" {
-		dockerRegistryOrg = o.GetDockerRegistryOrg(projectConfig, gitInfo)
-	}
-	appName := gitInfo.Name
-	return dockerRegistry + "/" + dockerRegistryOrg + "/" + appName
-}
-
-func (o *StepCreateTaskOptions) getDockerRegistry(projectConfig *config.ProjectConfig) string {
-	dockerRegistry := o.DockerRegistry
-	if dockerRegistry == "" {
-		dockerRegistry = o.GetDockerRegistry(projectConfig)
-	}
-	return dockerRegistry
-}
-
 func (o *StepCreateTaskOptions) getClientsAndNamespace() (tektonclient.Interface, jxclient.Interface, kubeclient.Interface, string, error) {
 	tektonClient, _, err := o.TektonClient()
 	if err != nil {

--- a/pkg/cmd/step/git/step_git.go
+++ b/pkg/cmd/step/git/step_git.go
@@ -7,8 +7,6 @@ import (
 	"github.com/spf13/cobra"
 
 	"github.com/jenkins-x/jx/pkg/cmd/opts"
-	"github.com/jenkins-x/jx/pkg/log"
-	"github.com/jenkins-x/jx/pkg/util"
 )
 
 // StepGitOptions contains the command line flags
@@ -46,57 +44,4 @@ func NewCmdStepGit(commonOpts *opts.CommonOptions) *cobra.Command {
 // Run implements this command
 func (o *StepGitOptions) Run() error {
 	return o.Cmd.Help()
-}
-
-func (o *StepGitOptions) dropRepositories(repoIds []string, message string) error {
-	var answer error
-	for _, repoId := range repoIds {
-		err := o.dropRepository(repoId, message)
-		if err != nil {
-			log.Logger().Warnf("Failed to drop repository %s: %s", util.ColorInfo(repoIds), util.ColorError(err))
-			if answer == nil {
-				answer = err
-			}
-		}
-	}
-	return answer
-}
-
-func (o *StepGitOptions) dropRepository(repoId string, message string) error {
-	if repoId == "" {
-		return nil
-	}
-	log.Logger().Infof("Dropping nexus release repository %s", util.ColorInfo(repoId))
-	err := o.RunCommand("mvn",
-		"org.sonatype.plugins:nexus-staging-maven-plugin:1.6.5:rc-drop",
-		"-DserverId=oss-sonatype-staging",
-		"-DnexusUrl=https://oss.sonatype.org",
-		"-DstagingRepositoryId="+repoId,
-		"-Ddescription=\""+message+"\" -DstagingProgressTimeoutMinutes=60")
-	if err != nil {
-		log.Logger().Warnf("Failed to drop repository %s due to: %s", repoId, err)
-	} else {
-		log.Logger().Infof("Dropped repository %s", util.ColorInfo(repoId))
-	}
-	return err
-}
-
-func (o *StepGitOptions) releaseRepository(repoId string) error {
-	if repoId == "" {
-		return nil
-	}
-	log.Logger().Infof("Releasing nexus release repository %s", util.ColorInfo(repoId))
-	options := o
-	err := options.RunCommand("mvn",
-		"org.sonatype.plugins:nexus-staging-maven-plugin:1.6.5:rc-release",
-		"-DserverId=oss-sonatype-staging",
-		"-DnexusUrl=https://oss.sonatype.org",
-		"-DstagingRepositoryId="+repoId,
-		"-Ddescription=\"Next release is ready\" -DstagingProgressTimeoutMinutes=60")
-	if err != nil {
-		log.Logger().Warnf("Failed to release repository %s due to: %s", repoId, err)
-	} else {
-		log.Logger().Infof("Released repository %s", util.ColorInfo(repoId))
-	}
-	return err
 }

--- a/pkg/cmd/step/helm/step_helm.go
+++ b/pkg/cmd/step/helm/step_helm.go
@@ -5,8 +5,6 @@ import (
 	"path/filepath"
 	"text/template"
 
-	"github.com/jenkins-x/jx/pkg/cmd/step/git/credentials"
-
 	"github.com/jenkins-x/jx/pkg/cmd/opts/step"
 
 	"github.com/ghodss/yaml"
@@ -18,8 +16,6 @@ import (
 	"github.com/jenkins-x/jx/pkg/cmd/helper"
 	"github.com/jenkins-x/jx/pkg/helm"
 	"github.com/spf13/cobra"
-
-	"os"
 
 	"github.com/jenkins-x/jx/pkg/cmd/opts"
 	"github.com/jenkins-x/jx/pkg/log"
@@ -83,134 +79,6 @@ func (o *StepHelmOptions) addStepHelmFlags(cmd *cobra.Command) {
 	cmd.Flags().BoolVarP(&o.https, "clone-https", "", true, "Clone the environment Git repo over https rather than ssh which uses `git@foo/bar.git`")
 	cmd.Flags().BoolVarP(&o.RemoteCluster, "remote", "", false, "If enabled assume we are in a remote cluster such as a stand alone Staging/Production cluster")
 	cmd.Flags().StringVarP(&o.GitProvider, "git-provider", "", "github.com", "The Git provider for the environment Git repository")
-}
-
-func (o *StepHelmOptions) dropRepositories(repoIds []string, message string) error {
-	var answer error
-	for _, repoId := range repoIds {
-		err := o.dropRepository(repoId, message)
-		if err != nil {
-			log.Logger().Warnf("Failed to drop repository %s: %s", util.ColorInfo(repoIds), util.ColorError(err))
-			if answer == nil {
-				answer = err
-			}
-		}
-	}
-	return answer
-}
-
-func (o *StepHelmOptions) dropRepository(repoId string, message string) error {
-	if repoId == "" {
-		return nil
-	}
-	log.Logger().Infof("Dropping helm release repository %s", util.ColorInfo(repoId))
-	err := o.RunCommand("mvn",
-		"org.sonatype.plugins:helm-staging-maven-plugin:1.6.5:rc-drop",
-		"-DserverId=oss-sonatype-staging",
-		"-DhelmUrl=https://oss.sonatype.org",
-		"-DstagingRepositoryId="+repoId,
-		"-Ddescription=\""+message+"\" -DstagingProgressTimeoutMinutes=60")
-	if err != nil {
-		log.Logger().Warnf("Failed to drop repository %s due to: %s", repoId, err)
-	} else {
-		log.Logger().Infof("Dropped repository %s", util.ColorInfo(repoId))
-	}
-	return err
-}
-
-func (o *StepHelmOptions) releaseRepository(repoId string) error {
-	if repoId == "" {
-		return nil
-	}
-	log.Logger().Infof("Releasing helm release repository %s", util.ColorInfo(repoId))
-	options := o
-	err := options.RunCommand("mvn",
-		"org.sonatype.plugins:helm-staging-maven-plugin:1.6.5:rc-release",
-		"-DserverId=oss-sonatype-staging",
-		"-DhelmUrl=https://oss.sonatype.org",
-		"-DstagingRepositoryId="+repoId,
-		"-Ddescription=\"Next release is ready\" -DstagingProgressTimeoutMinutes=60")
-	if err != nil {
-		log.Logger().Infof("Failed to release repository %s due to: %s", repoId, err)
-	} else {
-		log.Logger().Infof("Released repository %s", util.ColorInfo(repoId))
-	}
-	return err
-}
-
-func (o *StepHelmOptions) cloneProwPullRequest(dir, gitProvider string) (string, error) {
-
-	stepOpts := step.StepOptions{
-		CommonOptions: o.CommonOptions,
-	}
-	gitOpts := credentials.StepGitCredentialsOptions{
-		StepOptions: stepOpts,
-	}
-
-	err := gitOpts.Run()
-	if err != nil {
-		return "", fmt.Errorf("failed to create Git credentials file: %v", err)
-	}
-
-	org := os.Getenv(REPO_OWNER)
-	if org == "" {
-		return "", fmt.Errorf("no %s env var found", REPO_OWNER)
-	}
-
-	repo := os.Getenv(REPO_NAME)
-	if org == "" {
-		return "", fmt.Errorf("no %s env var found", REPO_NAME)
-	}
-
-	var gitURL string
-	if o.https {
-		gitURL = fmt.Sprintf("https://%s/%s/%s.git", gitProvider, org, repo)
-	} else {
-		gitURL = fmt.Sprintf("git@%s:%s/%s.git", gitProvider, org, repo)
-	}
-
-	err = o.RunCommand("git", "clone", gitURL)
-	if err != nil {
-		return "", err
-	}
-
-	prCommit := os.Getenv(PULL_PULL_SHA)
-	if org == "" {
-		return "", fmt.Errorf("no %s env var found", PULL_PULL_SHA)
-	}
-
-	if dir != "" {
-		dir = filepath.Join(dir, repo)
-	} else {
-		pwd, err := os.Getwd()
-		if err != nil {
-			return "", err
-		}
-		dir = filepath.Join(pwd, repo)
-	}
-
-	err = o.RunCommandFromDir(dir, "git", "checkout", prCommit)
-	if err != nil {
-		return "", err
-	}
-
-	err = o.RunCommand("cd", repo)
-	if err != nil {
-		return "", err
-	}
-
-	err = o.RunCommandFromDir(dir, "git", "checkout", "-b", "pr")
-	if err != nil {
-		return "", err
-	}
-	exists, err := util.DirExists(filepath.Join(dir, "env"))
-	if err != nil {
-		return "", err
-	}
-	if exists {
-		dir = filepath.Join(dir, "env")
-	}
-	return dir, nil
 }
 
 func (o *StepHelmOptions) discoverValuesFiles(dir string) ([]string, error) {

--- a/pkg/cmd/step/report/step_report_image_version.go
+++ b/pkg/cmd/step/report/step_report_image_version.go
@@ -375,25 +375,3 @@ func (o *StepReportImageVersionOptions) createStashContainer(volumeMounts []core
 		VolumeMounts: volumeMounts,
 	}
 }
-
-func (o *StepReportImageVersionOptions) getPackageVersion(name string) (string, error) {
-	args := []string{"version"}
-	switch name {
-	case "jx":
-		args = []string{"--version"}
-	case "kubectl":
-		args = append(args, "--client", "--short")
-	case "helm":
-		args = append(args, "--client", "--short")
-	case "helm3":
-		args = append(args, "--short")
-	}
-	version, err := o.GetCommandOutput("", name, args...)
-
-	// lets trim non-numeric prefixes such as for `git version` returning `git version 1.2.3`
-	idxs := numberRegex.FindStringIndex(version)
-	if len(idxs) > 0 {
-		return version[idxs[0]:], err
-	}
-	return version, err
-}

--- a/pkg/cmd/step/step_next_version.go
+++ b/pkg/cmd/step/step_next_version.go
@@ -413,23 +413,4 @@ func (o *StepNextVersionOptions) SetVersion() error {
 	return nil
 }
 
-func (o *StepNextVersionOptions) setPackageVersion(b []byte) error {
-	jsPackage := PackageJSON{}
-	err := json.Unmarshal(b, &jsPackage)
-	if err != nil {
-		return err
-	}
-	jsPackage.Version = o.NewVersion
-
-	return nil
-}
-
-func (o *StepNextVersionOptions) setChartVersion(b []byte) error {
-	return nil
-}
-
-func (o *StepNextVersionOptions) setPomVersion(b []byte) error {
-	return nil
-}
-
 // returns a string array containing the git owner and repo name for a given URL

--- a/pkg/cmd/step/verify/common.go
+++ b/pkg/cmd/step/verify/common.go
@@ -1,8 +1,6 @@
 package verify
 
 import (
-	"fmt"
-
 	"github.com/jenkins-x/jx/pkg/kube"
 	"github.com/jenkins-x/jx/pkg/log"
 	"github.com/jenkins-x/jx/pkg/util"
@@ -10,22 +8,6 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
-
-func (o *StepVerifyOptions) validateSecret(secretName, ns string) error {
-	kubeClient, _, err := o.KubeClientAndDevNamespace()
-	if err != nil {
-		return err
-	}
-	secret, err := kubeClient.CoreV1().Secrets(ns).Get(secretName, metav1.GetOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "could not find the Secret %s in the namespace: %s", secretName, ns)
-	}
-	if secret.Data == nil || len(secret.Data[secretName]) == 0 {
-		return fmt.Errorf("the Secret %s in the namespace: %s does not have a key: %s", secretName, ns, secretName)
-	}
-	log.Logger().Infof("external-dns is valid: there is a Secret: %s in namespace: %s\n", util.ColorInfo(secretName), util.ColorInfo(ns))
-	return nil
-}
 
 func (o *StepVerifyOptions) validateKaniko(ns string) error {
 	kubeClient, _, err := o.KubeClientAndDevNamespace()

--- a/pkg/cmd/step/verify/step_verify_environments.go
+++ b/pkg/cmd/step/verify/step_verify_environments.go
@@ -126,24 +126,6 @@ func (o *StepVerifyEnvironmentsOptions) Run() error {
 	return nil
 }
 
-func (o *StepVerifyEnvironmentsOptions) prDevEnvironment(gitRepoName string, environmentsOrg string, privateRepo bool, user *auth.UserAuth, requirements *config.RequirementsConfig, server *auth.AuthServer, createPr bool) error {
-	fromGitURL := os.Getenv(boot.ConfigRepoURLEnvVarName)
-	gitRef := os.Getenv(boot.ConfigBaseRefEnvVarName)
-
-	log.Logger().Debugf("Defined %s env variable value: %s", boot.ConfigRepoURLEnvVarName, fromGitURL)
-	log.Logger().Debugf("Defined %s env variable value: %s", boot.ConfigBaseRefEnvVarName, gitRef)
-
-	_, err := gits.ParseGitURL(fromGitURL)
-	if err != nil {
-		return err
-	}
-
-	log.Logger().Infof("The environment git repositories look good\n")
-	fmt.Println()
-
-	return nil
-}
-
 func (o *StepVerifyEnvironmentsOptions) storeRequirementsInTeamSettings(requirements *config.RequirementsConfig) error {
 	log.Logger().Infof("Storing the requirements in team settings in the dev environment\n")
 	err := o.ModifyDevEnvironment(func(env *v1.Environment) error {

--- a/pkg/cmd/sync/sync.go
+++ b/pkg/cmd/sync/sync.go
@@ -253,9 +253,3 @@ func (o *SyncOptions) CreateKsync(client kubernetes.Interface, ns string, name s
 
 	return o.RunCommand("ksync", "create", "--name", name, "-l", "jenkins.io/devpod="+name, reload, "-n", ns, dir, remoteDir)
 }
-
-func (o *SyncOptions) killWatchProcess(cmd *exec.Cmd) {
-	if err := cmd.Process.Kill(); err != nil {
-		log.Logger().Warnf("failed to kill 'ksync watch' process: %s", err)
-	}
-}

--- a/pkg/github/github_app.go
+++ b/pkg/github/github_app.go
@@ -25,17 +25,6 @@ type response struct {
 	AppName      string
 }
 
-func (gh *GithubApp) isGithubAppEnabled() (bool, error) {
-	requirementConfig, err := gh.getRequirementConfig()
-	if err != nil {
-		return false, err
-	}
-	if requirementConfig != nil && requirementConfig.GithubApp != nil {
-		return requirementConfig.GithubApp.Enabled, nil
-	}
-	return false, nil
-}
-
 // Install - confirms that the github app is installed and if it isn't then prints out a url for the user to install
 func (gh *GithubApp) Install(owner string, repo string, fileHandles util.IOFileHandles, newRepo bool) (bool, error) {
 	installed, accessToRepo, url, appName, err := gh.isInstalled(owner, repo)

--- a/pkg/gits/git_fake.go
+++ b/pkg/gits/git_fake.go
@@ -572,10 +572,6 @@ func (g *GitFake) LoadFileFromBranch(dir string, branch string, file string) (st
 	return "", nil
 }
 
-func (g *GitFake) notFound() error {
-	return fmt.Errorf("Not found")
-}
-
 // FetchUnshallow deepens a shallow git clone
 func (g *GitFake) FetchUnshallow(dir string) error {
 	return nil

--- a/pkg/helm/helm_cli.go
+++ b/pkg/helm/helm_cli.go
@@ -12,7 +12,6 @@ import (
 	"github.com/blang/semver"
 	"github.com/jenkins-x/jx/pkg/kube"
 
-	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/kubernetes/pkg/util/slice"
 
 	"github.com/jenkins-x/jx/pkg/log"
@@ -684,38 +683,6 @@ func addUsernamePasswordToURL(urlStr string, username string, password string) (
 		return u.String(), nil
 	}
 	return urlStr, nil
-}
-
-func (h *HelmCLI) getCurrentNamespace() (string, error) {
-	config, _, err := h.Kube().LoadConfig()
-	if err != nil {
-		return "", errors.Wrap(err, "loading Kubernetes configuration")
-	}
-	currentNS := kube.CurrentNamespace(config)
-
-	return currentNS, nil
-}
-
-func (h *HelmCLI) setNamespace(namespace string) error {
-	config, pathOptions, err := h.Kube().LoadConfig()
-	if err != nil {
-		return errors.Wrap(err, "loading Kubernetes configuration")
-	}
-
-	newConfig := *config
-	ctx := kube.CurrentContext(config)
-	if ctx == nil {
-		return fmt.Errorf("unable to get context")
-	}
-	if ctx.Namespace == namespace {
-		return nil
-	}
-	ctx.Namespace = namespace
-	err = clientcmd.ModifyConfig(pathOptions, newConfig, false)
-	if err != nil {
-		return fmt.Errorf("failed to update the kube config %s", err)
-	}
-	return nil
 }
 
 // extractSemanticVersion tries to extract a semantic version string substring from the specified string

--- a/pkg/helm/helm_template.go
+++ b/pkg/helm/helm_template.go
@@ -1067,27 +1067,6 @@ func (h *HelmTemplate) runKubectlWithOutput(args ...string) (string, error) {
 	return h.Runner.RunWithoutRetry()
 }
 
-// getChartNameAndVersion returns the chart name and version for the current chart folder
-func (h *HelmTemplate) getChartNameAndVersion(chartDir string, version *string) (string, string, error) {
-	versionText := ""
-	if version != nil && *version != "" {
-		versionText = *version
-	}
-	file := filepath.Join(chartDir, ChartFileName)
-	if !filepath.IsAbs(chartDir) {
-		file = filepath.Join(h.Runner.CurrentDir(), file)
-	}
-	exists, err := util.FileExists(file)
-	if err != nil {
-		return "", versionText, err
-	}
-	if !exists {
-		return "", versionText, fmt.Errorf("No file %s found!", file)
-	}
-	chartName, versionText, err := LoadChartNameAndVersion(file)
-	return chartName, versionText, err
-}
-
 // getChart returns the chart metadata for the given dir
 func (h *HelmTemplate) getChart(chartDir string, version string) (*chart.Metadata, string, error) {
 	file := filepath.Join(chartDir, ChartFileName)

--- a/pkg/tekton/syntax/pipeline.go
+++ b/pkg/tekton/syntax/pipeline.go
@@ -277,10 +277,6 @@ type PipelineOverride struct {
 
 var _ apis.Validatable = (*ParsedPipeline)(nil)
 
-func (s *Stage) taskName() string {
-	return strings.ToLower(strings.NewReplacer(" ", "-").Replace(s.Name))
-}
-
 // stageLabelName replaces invalid characters in stage names for label usage.
 func (s *Stage) stageLabelName() string {
 	return MangleToRfc1035Label(s.Name, "")
@@ -2132,18 +2128,6 @@ func (ts *transformedStage) getEnclosing(depth int8) *transformedStage {
 		return nil
 	} else {
 		return ts.EnclosingStage.getEnclosing(depth)
-	}
-}
-
-// Return the first stage that will execute before this stage
-// Depth must be >= 0
-func (ts transformedStage) getClosestAncestor() *transformedStage {
-	if ts.PreviousSiblingStage != nil {
-		return ts.PreviousSiblingStage
-	} else if ts.EnclosingStage == nil {
-		return nil
-	} else {
-		return ts.EnclosingStage.getClosestAncestor()
 	}
 }
 

